Mathlib/LinearAlgebra/FreeModule/Finite/Rank.lean

```diff
@@ -651,12 +652,13 @@ variable {R V}
 
 theorem Submodule.bot_eq_top_of_rank_eq_zero [NoZeroSMulDivisors R V] (h : Module.rank R V = 0) :
     (⊥ : Submodule R V) = ⊤ := by
+  nontriviality R
   rw [rank_zero_iff] at h
   exact Subsingleton.elim _ _
 #align bot_eq_top_of_rank_eq_zero Submodule.bot_eq_top_of_rank_eq_zero
 
 @[simp]
-theorem Submodule.rank_eq_zero [NoZeroSMulDivisors R V] {S : Submodule R V} :
+theorem Submodule.rank_eq_zero [Nontrivial R] [NoZeroSMulDivisors R V] {S : Submodule R V} :
```

Comment:
Shouldn't this also be true for the zero ring? Or does that give `Module.rank R S = 1`?
If so, please leave a comment.

Reply:
Yes, {0} is a maximal linear independent set in a vector space over the zero ring, which has cardinality 1.

Reply:
I think the way to think about it is that the only module over the zero ring is free of arbitrary dimension. But since we take the sup over all subsets rather than families indexed by arbitrary types, the cardinality is bounded by the cardinality of the module, which is 1.